test/test_nn.py

```diff
@@ -10178,16 +10184,24 @@ def get_grid(device='cpu', data=None):
                         lambda inp, grid: F.grid_sample(inp, grid, mode=mode, padding_mode=padding_mode,
                                                         align_corners=align_corners),
                         (input, grid)))
+                    input = input.requires_grad_(False)
```

Comment:
Why is this one special cased here?

Reply:
I think we want to check whether the grid gradient remains correct when we are in the special case where input.requires_grad() is False

Reply:
Don't we already run gradcheck above for both input grad being True and False?

Reply:
Well this was to-mi code for the 2D case, but I don't see which line you are referencing

Reply:
What I meant is that this could be merged with the test above that checks the case input=True more explicitly. But maybe that would make the code look weird?

Reply:
I moved those gradcheck tests below, in the loop that tests for input_requires_grad in [True, False]